src/imap.rs

```diff
@@ -405,6 +406,28 @@ impl Default for ImapConfig {
     }
 }
 
+    fn body_structure_is_autocrypt_setup(bs: &imap_proto::types::BodyStructure)->bool
+    {
+        match bs
+        {
+            imap_proto::BodyStructure::Text{..} => return false,
+            // imap_proto::BodyStructure::Text{common, other, lines, extension} => return false,
+            imap_proto::BodyStructure::Message{..} => return false,
+            // imap_proto::BodyStructure::Message{common, other, envelope, body, lines, extension} => return false,
+            imap_proto::BodyStructure::Basic{common, ..} =>
+            {
+                return common.ty.ty == "application" && common.ty.subtype == "autocrypt-setup";
+            },
+            imap_proto::BodyStructure::Multipart{bodies, ..} =>  {
+                for bs in bodies
```

Comment:
try to simplify to:
```rust
return bodies.any(|bs| body_structure_is_autocrypt_setup(&bs))
```
or this might even work:

```rust
return bodies.any(body_structure_is_autocrypt_setup)
```